### PR TITLE
Include CPU usage in partitioning

### PR DESCRIPTION
**Pull request by mundya**
This is a slightly conservative estimate of CPU usage based on the results in "An efficient SpiNNaker implementation of the NEF" [IJCNN, 2015].

### CODE REVIEW

File: nengo_spinnaker/operators/lif.py [hunk 1]
```diff
@@ -135,14 +135,21 @@ def make_vertices(self, model, n_steps):  # TODO remove n_steps
...
         # TODO: Include CPU usage constraint
+        def cpu_usage(sl):
+            n_neurons = (sl.stop - sl.start)
+            return (245 + 43*self.ensemble.size_in + 100 + 702*size_out +
+                    188 + 69*n_neurons + 13*n_neurons*self.ensemble.size_in)
+
         self.vertices = list()
         sdram_constraint = partition.Constraint(8*2**20)  # Max 8MiB
         dtcm_constraint = partition.Constraint(64*2**10, .75)  # 75% of 64KiB
+        cpu_constraint = partition.Constraint(200000, .80)
```
Comment (tcstewar) [inline nengo_spinnaker/operators/lif.py]: A comment here that this is 80% of 200000 cycles would be nice

Comment (mundya) [inline nengo_spinnaker/operators/lif.py]: Fixed.

File: nengo_spinnaker/operators/lif.py [hunk 2]
```diff
@@ -135,14 +135,21 @@ def make_vertices(self, model, n_steps):  # TODO remove n_steps
         # can be quite lax to allow for lots of data probing.
         # TODO: Include other DTCM usage
         # TODO: Include CPU usage constraint
+        def cpu_usage(sl):
+            n_neurons = (sl.stop - sl.start)
+            return (245 + 43*self.ensemble.size_in + 100 + 702*size_out +
+                    188 + 69*n_neurons + 13*n_neurons*self.ensemble.size_in)
+
```
Comment (tcstewar) [inline nengo_spinnaker/operators/lif.py]: A comment here about a resource for those numbers would be handy (perhaps a reference to the paper?)

File: nengo_spinnaker/operators/lif.py [hunk 3]
```diff
@@ -135,14 +135,21 @@ def make_vertices(self, model, n_steps):  # TODO remove n_steps
         # can be quite lax to allow for lots of data probing.
         # TODO: Include other DTCM usage
         # TODO: Include CPU usage constraint
```
Comment (tcstewar) [inline nengo_spinnaker/operators/lif.py]: I assume this TODO should be removed?

**Comment by mundya** [comment]
@tcstewar  How's it now? Thanks :)

**Comment by tcstewar** [comment]
Looks great to me!  Merging....

**Comment by mundya** [comment]
Cool, I'll push a new release to PyPi.

**Pull request merged**